strstr/host/strstr_host.c

```diff
@@ -0,0 +1,147 @@
+/* Communication with a DPU via the MRAM. */
+/* Populate the MRAM with a collection of bytes and request the DPU to */
+/* compute the checksum. */
+
+#include <dpu.h>
+#include <assert.h>
+#include <stdint.h>
+#include <stdio.h>
+#include "common.h"
+#include <dpu_log.h>
+#include <time.h>
+#include <string.h>
+
+#ifndef DPU_BINARY
+#define DPU_BINARY "build/strstr_dpu"
+#endif
+
+#define DPU_LOG_ENABLE
+#define CYCLES 500000000
+
+char* read_json_file() {
+    char *source = NULL;
+    FILE *fp = fopen("business.json", "r");
+    if (fp != NULL) {
```

Comment:
This logic is correct, but confusing. What happens if opening the file ever fails? It will continue executing after the 'if', which is probably a bug. Instead, you can check if (fp == NULL), and just exit right away.